jupyterhub/templates/image-puller/_helpers-daemonset.tpl

```diff
@@ -70,6 +70,11 @@ spec:
               {{- include "jupyterhub.userNodeAffinityRequired" . | nindent 14 }}
       {{- end }}
       terminationGracePeriodSeconds: 0
+      {{- if .hook }}
+      serviceAccountName: "hook-image-puller-serviceaccount"
+      {{- else }}
+      serviceAccountName: "continuous-image-puller-serviceaccount"
+      {{- end }}
```

Comment:
The introduction of hardcoded names can cause issues in some edge case installation, so we have avoided it. Can you mimic how other resources have been named?

Reply:
The name is set the same no matter what, this is the issue, so its done differently from how other resources get their name.

The daemonset name is for comparison named in this way:

name: {{ include "jupyterhub.hook-image-puller.fullname" . }}

Reply:
Thanks. 

I already updated the code and created a new entry, `.Values.prePuller.hook.serviceAccountImagePuller`. I did this so the users can have more flexibility defining serviceAccounts. 

Another alternative is to keep just one `.Values.prePuller.hook.serviceAccount` and do something similar to what is done to the daemonset name (by appending for example -image-puller at the end of serviceAccount)

Which approach do you prefer? I can change it to the second one if you like it more :)

Reply:
Lets keep them separate, because this new one isn't supposed to be granted any permissions, while the one for the image-awaiter needs permissions inspecting daemonsets for example.